lib/hocon/impl/config_reference.rb

```diff
@@ -72,4 +73,56 @@ def resolve_substitutions(context, source)
     end
 
   end
+
+  def self.not_resolved
+    error_message = "need to Config#resolve, see the API docs for Config#resolve; substitution not resolved: #{self}"
+    Hocon::ConfigError::ConfigNotResolvedError.new(error_message, nil)
+  end
+
+  def value_type
+    raise self.class.not_resolved
+  end
+
+  def unwrapped
+    raise self.class.not_resolved
+  end
+
+  def new_copy(new_origin)
+    Hocon::Impl::ConfigReference.new(new_origin, @expr, @prefix_length)
+  end
+
+  def ignores_fallbacks
+    false
+  end
+
+  def resolve_status
+    Hocon::Impl::ResolveStatus::UNRESOLVED
+  end
+
+  def relativized(prefix)
+    new_expr = @expr.change_path(@expr.path,prepend(prefix))
```

Comment:
`change_path` only takes one argument ... maybe that comma is supposed to be a dot?